pennylane/transforms/metric_tensor.py

```diff
@@ -197,7 +267,19 @@ def qnode_execution_wrapper(self, qnode, targs, tkwargs):
     cjac_fn = qml.transforms.classical_jacobian(qnode, expand_fn=_expand_fn)
 
     def wrapper(*args, **kwargs):
-        mt = mt_fn(*args, **kwargs)
+        try:
+            mt = mt_fn(*args, **kwargs)
+        except qml.wires.WireError as e:
+            warnings.warn(
+                "A wire was not found while executing tapes for the metric tensor. "
+                "Probably you tried to use Hadamard tests and the device does not provide an "
+                "additional wire or the requested auxiliary wire does not exist on the device. "
+                "Reverting to the block-diagonal approximation. It will often be much more "
+                "efficient to request the block-diagonal approximation directly! "

```

Comment:
```suggestion
                "An auxiliary wire is not available."
                "\n\nThis can occur when computing the full metric tensor via the "
                "Hadamard test, and the device does not provide an "
                "additional wire or the requested auxiliary wire does not exist "
                "on the device."
                "\n\nReverting to the block-diagonal approximation. It will often be "
                "much more efficient to request the block-diagonal approximation directly!"
```